File: leetcode/c++/LCR_173_que_shi_de_shu_zi_lcof.cpp
// https://leetcode.cn/problems/que-shi-de-shu-zi-lcof/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int takeAttendance(vector<int>& records) {
        if (records[0] == 1)
            return 0;
        int i = 0, j = records.size() - 1;
        while (i <= j) {
            int mid = i + (j - i) / 2;
            if (records[mid] == mid)
                i = mid + 1;
            else
                j = mid - 1;
        }
        return i;
    }
};

// 有序数组一定要二分，O(logn)，O(1)

File: leetcode/golang/LCR_173_que_shi_de_shu_zi_lcof.go
func takeAttendance(records []int) int {
    if records[0] == 1 {
        return 0
    }
    i, j := 0, len(records) - 1
    for i <= j {
        m := i + (j-i) / 2
        if records[m] == m {
            i = m + 1
        } else {
            j = m - 1
        }
    }
    return i
}